csharp/src/SeedLang/Interpreter/Compiler.cs

```diff
@@ -21,11 +21,21 @@
 namespace SeedLang.Interpreter {
   // The compiler to convert an AST tree to bytecode.
   internal class Compiler : AstWalker {
+    // The class to collect comparison information during compilation. The information includes
+    // operators and registers to hold the operand values, which is used to generate comparison
+    // notifications.
+    private class ComparisonInfo {
+      public uint FirstId;
```

Comment:
1. The following value IDs are named `ValueIds`. Better to rename this to `FirstValueId` to make them consistent.

3. Although I figured out what `firstId` is for, I think it'd be better to add some very simple comments to let other developers know that it's for holding the first value ID of a comparison chain, such as `3 > 2 > 1 > 4`